apps/veyfi/components/ManageLockTab.tsx

```diff
@@ -0,0 +1,154 @@
+
+	const web3Provider = provider as ethers.providers.Web3Provider;
+	const userAddress = address as TAddress;
+	const hasLockedAmount = formatBN(positions?.deposit?.balance).gt(0);
+	const willExtendLock = formatBN(lockTime).gt(0);
+	const timeUntilUnlock = positions?.unlockTime ? getTimeUntil(positions?.unlockTime) : undefined;
+	const weeksToUnlock = timeUntilUnlock ? toWeeks(timeUntilUnlock) : 0;
+	const newUnlockTime = toTime(positions?.unlockTime) + fromWeeks(toTime(lockTime));
+	const hasPenalty = formatBN(positions?.penalty).gt(0);
+
+	const votingPower = useMemo((): BigNumber => {
+		if(!positions?.deposit || !newUnlockTime) {
+			return formatBN(0);
+		}
+		return willExtendLock ? getVotingPower(positions?.deposit?.underlyingBalance,  newUnlockTime) : formatBN(positions?.deposit?.balance);
+	}, [positions?.deposit, newUnlockTime, willExtendLock]);
+	
+	const {isValid: isValidLockTime, error: lockTimeError} = validateAmount({
+		amount: lockTime,
+		minAmountAllowed: MIN_LOCK_TIME
+	});
+
+	const {isValid: isValidNetwork} = validateNetwork({supportedNetwork: 1, walletNetwork: chainID});
+
+	const executeExtendLockTime = (): void => {
+		if (!votingEscrow || !userAddress) {
+			return;
+		}
+		extendLockTime(web3Provider, userAddress, votingEscrow.address, toSeconds(newUnlockTime));
+	};
+	
+	const executeWithdrawLocked = (): void => {
+		if (!votingEscrow  || !userAddress) {
+			return;
+		}
+		withdrawLocked(web3Provider, userAddress, votingEscrow.address);
+	};
+	
+	return ( 
+		<div className={'grid grid-cols-1 gap-6 md:grid-cols-2 md:gap-16'}>
+			<div className={'col-span-1 grid w-full gap-6'}>
+				<div className={'md:min-h-[104px]'}>
+					<h2 className={'m-0 text-2xl font-bold'}>
+						{'Extend lock'}
+					</h2>
+					<div className={'mt-6 text-neutral-600'} >
+						<p>{'Want to lock for longer? Extend your lock period to increase your gauge boost weight.'}</p>
+					</div>
+				</div>
+				<div className={'grid grid-cols-1 gap-6 md:grid-cols-2'}>
+					<AmountInput
+						label={'Current lock period (weeks)'}
+						amount={weeksToUnlock}
+						disabled
+					/>
+					<AmountInput
+						label={'Increase lock period (weeks)'}
+						amount={lockTime}
+						onAmountChange={(amount): unknown => set_lockTime(Math.floor(toTime(amount)).toString())}
+						maxAmount={MAX_LOCK_TIME - weeksToUnlock > 0 ? MAX_LOCK_TIME - weeksToUnlock : 0}
+						disabled={!hasLockedAmount}
+						error={lockTimeError}
+						legend={'min 1'}
+					/>
+				</div>
+				<div className={'grid grid-cols-1 gap-6 md:grid-cols-2 md:pb-5'}>
+					<AmountInput
+						label={'Total veYFI'}
+						amount={formatUnits(votingPower, 18)}
+						disabled
+					/>
+					<Button
+						className={'w-full md:mt-7'}
+						onClick={executeExtendLockTime}
+						isBusy={extendLockTimeStatus.loading}
+						disabled={!isActive || !isValidNetwork || !isValidLockTime || extendLockTimeStatus.loading}
```

Comment:
Wondering if we could extract this to an external function to have all this logic in a single place, there seems to be a lot of conditions to have the buttons disabled in many places.